inspirehep/records/indexer/tasks.py

```
@@ -0,0 +1,122 @@
+            f"Current version: {record.model.version_id}."
+        )
+        raise StaleDataError()
+    return record
+
+
+@shared_task(ignore_result=False, bind=True)
+def process_references_for_record(
+    pid_type=None, pid_value=None, record_version=None, record=None
+):
+    if not record:
+        record = get_record(pid_type, pid_value, record_version)
+    pids = record.get_modified_references()
+
+    if not pids:
+        logger.debug(f"No references change for record {pid_type}:{pid_value}")
+        return None
+    logger.debug(
+        f"({pid_value}) There are {len(pids)} records where references changed"
+    )
+    uuids = [
+        str(pid.object_uuid)
+        for pid in db.session.query(PersistentIdentifier.object_uuid).filter(
+            PersistentIdentifier.object_type == "rec",
+            tuple_(PersistentIdentifier.pid_type, PersistentIdentifier.pid_value).in_(
+                pids
+            ),
+        )
+    ]
+
+    if uuids:
+        logger.info(f"({pid_value}) contains pids - starting batch")
+        return batch_index(uuids)
+
+    else:
+        raise MissingCitedRecordError(
+            f"Cited records to reindex not found:\nuuids: {uuids}"
+        )
+
+
+@shared_task(ignore_result=False, bind=True, max_retries=6)
+def index_record(self, pid_value, pid_type, record_version=None, deleted=None):
+    logger.info(
+        f"Starting shared task `index_record` for "
+        f"record {pid_type}:{pid_value}:v{record_version}"
+    )
+    try:
+        record = get_record(pid_type, pid_value, record_version)
+    except (NoResultFound, StaleDataError) as e:
+        logger.warn(
+            f"Record {pid_type}:{pid_value} not yet at version {record_version} on DB"
+        )
+        backoff = 2 ** (self.request.retries + 1)
+        if self.max_retries < self.request.retries + 1:
+            logger.warn(f"({pid_value}) - Failing - too many retries")
```

Should we `raise` again in this case? Not sure tough :)

---

I think that ending this with exception is more natural than failing silently.